ot/da.py

```diff
@@ -843,10 +909,807 @@ def predict(self, x):
+            # Simple optimisation to gain speed (inspect is slow)
+            return self
+        valid_params = self.get_params(deep=True)
+        # for key, value in iteritems(params):
+        for key, value in params.items():
+            split = key.split('__', 1)
+            if len(split) > 1:
+                # nested objects case
+                name, sub_name = split
+                if name not in valid_params:
+                    raise ValueError('Invalid parameter %s for estimator %s. '
+                                     'Check the list of available parameters '
+                                     'with `estimator.get_params().keys()`.' %
+                                     (name, self))
+                sub_object = valid_params[name]
+                sub_object.set_params(**{sub_name: value})
+            else:
+                # simple objects case
+                if key not in valid_params:
+                    raise ValueError('Invalid parameter %s for estimator %s. '
+                                     'Check the list of available parameters '
+                                     'with `estimator.get_params().keys()`.' %
+                                     (key, self.__class__.__name__))
+                setattr(self, key, value)
+        return self
+
+    def __repr__(self):
+        from sklearn.base import _pprint
+        class_name = self.__class__.__name__
+        return '%s(%s)' % (class_name, _pprint(self.get_params(deep=False),
+                                               offset=len(class_name),),)
+
+    # __getstate__ and __setstate__ are omitted because they only contain
+    # conditionals that are not satisfied by our objects (e.g.,
+    # ``if type(self).__module__.startswith('sklearn.')``.
+
+
+def distribution_estimation_uniform(X):
+    """estimates a uniform distribution from an array of samples X
+
+    Parameters
+    ----------
+    X : array-like of shape = (n_samples, n_features)
+        The array of samples
+    Returns
+    -------
+    mu : array-like, shape = (n_samples,)
+        The uniform distribution estimated from X
+    """
+
+    return unif(X.shape[0])
+
+
+class BaseTransport(BaseEstimator):
+
+    def fit(self, Xs=None, ys=None, Xt=None, yt=None):
+        """Build a coupling matrix from source and target sets of samples
+        (Xs, ys) and (Xt, yt)
+        Parameters
+        ----------
+        Xs : array-like of shape = (n_source_samples, n_features)
+            The training input samples.
+        ys : array-like, shape = (n_source_samples,)
+            The class labels
+        Xt : array-like of shape = (n_target_samples, n_features)
+            The training input samples.
+        yt : array-like, shape = (n_labeled_target_samples,)
+            The class labels
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+
+        # pairwise distance
+        self.Cost = dist(Xs, Xt, metric=self.metric)
```

Comment:
any attribute changed in fit function should have trailing _
and caps should be avoided un var names. So I would write:

self.cost_ = dist(Xs, Xt, metric=self.metric)